hydroflow/tests/surface_codegen.rs

```diff
@@ -415,6 +416,75 @@ pub fn test_groupby() {
     println!();
 }
 
+#[test]
+pub fn test_demux_1() {
+    enum Shape {
+        Circle(f64),
+        Rectangle { width: f64, height: f64 },
+        Square(f64),
+    }
+
+    let mut df = hydroflow_syntax! {
+        demux = recv_iter([
+            Shape::Circle(5.0),
+            Shape::Rectangle { width: 10.0, height: 8.0 },
+            Shape::Square(9.0),
+        ]) -> demux(|shape, tl!(circ, rect)| {
```

Comment:
tl! is a macro of your creation yes? If we're exposing it to users, I'd prefer a more descriptive name.

Reply:
Will do this separately in #264 since this shows up in a good few other places